lib/auth/auth_with_roles.go

```diff
@@ -192,15 +193,32 @@ func (a *ServerWithRoles) actionWithExtendedContext(namespace, kind, verb string
 }
 
 // actionForKindSession is a special checker that grants access to session
-// recordings.  It can allow access to a specific recording based on the
+// recordings. It can allow access to a specific recording based on the
 // `where` section of the user's access rule for kind `session`.
-func (a *ServerWithRoles) actionForKindSession(namespace string, sid session.ID) error {
+func (a *ServerWithRoles) actionForKindSession(namespace string, sid session.ID) (types.SessionKind, error) {
```

Comment:
nit: is it possible to separate the calculation of the session kind from `actionXXX` function?

Reply:
We could isolate this, but it would require querying for the session end event again, as we only "save" the retrieved event for access check (using their context). Or we could retrieve it beforehand and pass it (and the fetch error) to this function. Either way, we would do additional work (or add more complexity to the `StreamSessionEvents`) if we don't want to fetch the sessions when they are not required (non-user streaming).

The way we're doing it now is we only have one retrieve (which is called only when RBAC is necessary, meaning we need to audit the access) and reuse the event the access checker used.